packages/debugger/src/service.ts

```diff
@@ -332,6 +332,22 @@ export class DebuggerService implements IDebugger, IDisposable {
     this._model.variables.scopes = variableScopes;
   }
 
+  async displayModules(): Promise<void> {
+    if (!this.session) {
+      throw new Error('No active debugger session');
+    }
+
+    const modules = await this.session.sendRequest('modules', {});
+    this._model.kernelSources.kernelSources = modules.body.modules.map(
+      module => {
+        return {
+          name: Object.keys(module)[0],
+          path: Object.values(module)[0]
```

Comment:
I was not able to test it locally with [commit 57a0fce](https://github.com/datalayer-externals/jupyter-kernel-python/tree/feat/debugModulesRequest) because the reply is a list of dictionaries. So this leads to name equals `id` and path equals _<id value>_.

Example of reply:

```json5
{
	"body": {
		"modules": [
			{
				"id": 0,
				"name": "__main__",
				"path": "/tmp/ipykernel_51649/730864761.py",
				"package": "None"
			},
			{
				"id": 1,
				"name": "IPython.core.interactiveshell",
				"path": ".../lib/python3.9/site-packages/IPython/core/interactiveshell.py",
				"package": "IPython.core"
			},
			{
				"id": 2,
				"name": "IPython.core.async_helpers",
				"path": ".../lib/python3.9/site-packages/IPython/core/async_helpers.py",
				"package": "IPython.core"
			},
			{
				"id": 3,
				"name": "ipykernel.zmqshell",
				"path": ".../lib/python3.9/site-packages/ipykernel/zmqshell.py",
				"package": "ipykernel"
			},
                        // ...
		]
	}
}
```

Reply:
I have double checked and it works fine on my env. Can you install ipykernel with `pip install git+https://github.com/datalayer-externals/jupyter-kernel-python@feat/debugModulesRequest#egg=ipykernel` and try again?

Reply:
:+1: it works

Reply:
Following up on this in https://github.com/jupyterlab/jupyterlab/issues/11966. Two thoughts:
- if we can support the old format (=do not require users to upgrade ipykernel etc), let's support it
- is the new format really valid in DAP?